src/System.Globalization/tests/TextElementEnumerator/TextElementEnumeratorTests.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class TextElementEnumeratorTests
+    {
+        [Fact]
+        public void Enumerate()
+        {
+            // Creates and initializes a string containing the following:
+            //   - a surrogate pair (high surrogate U+D800 and low surrogate U+DC00)
+            //   - a combining character sequence (the Latin small letter "a" followed by the combining grave accent)
+            //   - a base character (the ligature "")
+            string[] expectedElements = new string[] { "\uD800\uDC00", "\uD800\uDC00", "\u0061\u0300", "\u0061\u0300", "\u00C6" };
+            int[] expectedElementIndices = new int[] { 0, 2, 4 };
+            TextElementEnumerator enumerator = StringInfo.GetTextElementEnumerator("\uD800\uDC00\u0061\u0300\u00C6");
+
+            int counter = 0;
+            while (enumerator.MoveNext())
+            {
+                string currentTextElement = enumerator.GetTextElement();
+                Assert.Equal(expectedElements[enumerator.ElementIndex], currentTextElement);
+                Assert.Equal(currentTextElement, enumerator.Current);
+
+                Assert.Equal(expectedElementIndices[counter], enumerator.ElementIndex);
+                counter++;
+            }
+            Assert.Equal(expectedElementIndices.Length, counter);
+
+            enumerator.Reset();
+            counter = 0;
+            while (enumerator.MoveNext())
+            {
+                string currentTextElement = enumerator.GetTextElement();
+                Assert.Equal(expectedElements[enumerator.ElementIndex], currentTextElement);
+                Assert.Equal(currentTextElement, enumerator.Current);
+
+                Assert.Equal(expectedElementIndices[counter], enumerator.ElementIndex);
+                counter++;
+            }
+            Assert.Equal(expectedElementIndices.Length, counter);
```

Comment:
Why is this code duplicated?  Why not just put it in a loop of two iterations? e.g.
```C#
for (int i = 0; i < 2; i++)
{
    int counter = 0;
    while (enumerator.MoveNext())
    {
        ...
    }
    enumerator.Reset();
}
```